core/state/state_object.go

```diff
@@ -54,6 +54,28 @@ func (s Storage) Copy() Storage {
 	return cpy
 }
 
+// BalanceChangeReason is used to indicate the reason for a balance change, useful
+// for tracing and reporting.
+type BalanceChangeReason byte
```

Comment:
Perhaps move this and the constants below into some new file, like `core/state/metadata.go` ?